Add E2E tests to test error reporting on UI for invalid PipelineRun definition

zakisk opened the pull request:
adds E2E tests for gitea and github to ensure that if user supplies invalid PipelineRun definition, PAC should report errors on UI.

it was created as bug on JIRA but I confirmed it is working as expected. 
https://issues.redhat.com/browse/SRVKP-5776

savitaashture commented on test/github_pullrequest_test.go:
```diff
@@ -135,6 +137,57 @@ func TestGithubPullRequestSecondBadYaml(t *testing.T) {
 
+// TestGithubPullRequestInvalidSpecValues tests invalid field values of a PipelinRun and
+// ensures that these validation errors are reported on UI.
+func TestGithubPullRequestInvalidSpecValues(t *testing.T) {
```
@zakisk lets not run all tests on every PR otherwise we might hit a rate limit
Please add this condition
```
	if os.Getenv("NIGHTLY_E2E_TEST") != "true" {
		t.Skip("Skipping test since only enabled for nightly")
	}
```
which will just run on nightly 
ref : https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/github_pullrequest_privaterepository_test.go#L20-L22

zakisk replied:
is this test only for nightly?

savitaashture replied:
Yes

savitaashture commented on test/github_pullrequest_test.go:
```diff
@@ -135,6 +137,57 @@ func TestGithubPullRequestSecondBadYaml(t *testing.T) {
+func TestGithubPullRequestInvalidSpecValues(t *testing.T) {
+	ctx := context.Background()
+	g := &tgithub.PRTest{
+		Label:            "Github Rerequest",
```
This is not Rerequest right :thinking:

zakisk replied:
yes, will rename it.

savitaashture commented on test/github_pullrequest_test.go:
```diff
@@ -135,6 +137,57 @@ func TestGithubPullRequestSecondBadYaml(t *testing.T) {
+	// and digits ([\d]) patterns are used.
+	pattern := `invalid value: ([\d\w]+) should be <= pipeline duration: ([\w\.\/\-]+)(?:, ([\w\.\/\-]+))?` +
+		`|invalid value: ([\d\w]+) \+ ([\d\w]+) should be <= pipeline duration: ([\w\.\/\-]+), ([\w\.\/\-]+)` +
+		`|invalid value: ([\d\w]+) should be <= pipeline duration: ([\w\.\/\-]+)`
```
Can you put the pattern in common place so that can be referred from both GitHub and Gitea

zakisk replied:
may it be put here in [options.go](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/pkg/options/options.go)? or do I make a new folder in ./test/pkg?

savitaashture replied:
I think we can put under options.go

savitaashture commented on test/gitea_test.go:
```diff
@@ -134,6 +134,21 @@ func TestGiteaBadYaml(t *testing.T) {
 		"pipelinerun.*has failed.*expected exactly one, got neither: spec.pipelineRef, spec.pipelineSpec"), 10, "controller", github.Int64(20)))
 }
 
+// TestGiteaInvalidValues tests invalid field values of a PipelinRun and ensures that these
```
```suggestion
// TestGiteaInvalidSpecValues tests invalid field values of a PipelinRun and ensures that these
```

zakisk committed:
Add E2E tests to test error reporting on UI for invalid PR Yaml

adds E2E tests for gitea and github to ensure that if user supplies
invalid PipelineRun definition, PAC should report errors on UI.

it was created as bug on JIRA but I confirmed it is working as expected.
https://issues.redhat.com/browse/SRVKP-5776

Signed-off-by: Zaki Shaikh <zashaikh@redhat.com>

savitaashture commented:
/test

zakisk commented:
/test
